oozie/src/main/java/org/apache/falcon/oozie/feed/FeedRetentionWorkflowBuilder.java

```diff
@@ -51,20 +51,26 @@ public FeedRetentionWorkflowBuilder(Feed entity) {
     @Override public Properties build(Cluster cluster, Path buildPath) throws FalconException {
         WORKFLOWAPP workflow = new WORKFLOWAPP();
         String wfName = EntityUtil.getWorkflowName(Tag.RETENTION, entity).toString();
-
         //Add eviction action
         ACTION eviction = unmarshalAction(EVICTION_ACTION_TEMPLATE);
-        addTransition(eviction, SUCCESS_POSTPROCESS_ACTION_NAME, FAIL_POSTPROCESS_ACTION_NAME);
-        workflow.getDecisionOrForkOrJoin().add(eviction);
 
-        //Add post-processing actions
-        ACTION success = getSuccessPostProcessAction();
-        addTransition(success, OK_ACTION_NAME, FAIL_ACTION_NAME);
-        workflow.getDecisionOrForkOrJoin().add(success);
 
-        ACTION fail = getFailPostProcessAction();
-        addTransition(fail, FAIL_ACTION_NAME, FAIL_ACTION_NAME);
-        workflow.getDecisionOrForkOrJoin().add(fail);
+        if (!isPostProcessingEnabled()){
```

Comment:
Sorry from the last comment I understood I have to move the Boolean check to the parent class,which I did.

addTransition method is already in the parent class , can you please be bit more specific weather I have to move whole if/else condition to parent class or only if or something else.